tfjs-core/src/backends/webgl/glsl_version.ts

```diff
@@ -50,10 +50,22 @@ export function getGlslDifferences(): GLSL {
     texture2D = 'texture';
     output = 'outputColor';
     defineOutput = 'out vec4 outputColor;';
+
+    // Use custom isnan definition to work across differences between
+    // implementations on various platforms. While this should happen in ANGLE
+    // we still see differences between android and windows (on chrome) when
+    // using isnan directly.
     defineSpecialNaN = `
       bool isnan_custom(float val) {
-        return (val > 0. || val < 0. || val == 0.) ? false : true;
```

Comment:
that's crazy...

Reply:
close in form but slightly different to what we had before!